app/src/main/java/io/appium/uiautomator2/utils/AXWindowHelpers.java

```diff
@@ -99,20 +104,43 @@ private static AccessibilityNodeInfo[] getWindowRoots() {
         return result.toArray(new AccessibilityNodeInfo[0]);
     }
 
+    private static AccessibilityNodeInfo getTopmostWindowRootFromActivePackage() {
+        CharSequence activeRootPackageName = getActiveWindowRoot().getPackageName();
+        List<AccessibilityWindowInfo> windows = getWindows();
+
+        Optional<AccessibilityNodeInfo> topmostWindowRootFromActivePackage = windows.stream()
+                .filter(window -> window.getRoot() != null)
+                .filter(window -> window.getRoot().getPackageName().toString().contentEquals(activeRootPackageName))
```

Comment:
Object.equals(window.getRoot().getPackageName(), activeRootPackageName) would probably be safer?

Reply:
Agree. For `null` value handling.

Reply:
Addressed in 76836d98af70fc05070bab642303c13d4a1bc176.